lib/metrics.py

```diff
@@ -327,12 +328,19 @@ def _get_db_conn(self):
             self.db = None
 
         if not self.db:
+            # get_database config may return None or empty
             db_config = database_config.get_database_config()
+            if not db_config or "DatabaseType" not in db_config:
+                raise Exception(
```

Comment:
As previous, it's generally better to raise a RuntimeError as opposed to just an exception.

Although, here, a ValueError is probably better.

Reply:
Or even a custom, domain exception.

Reply:
Even better.

```
class MissingDatabaseConfigException(ValueError):
    pass

if not 'foo' in 'bar':
    raise MissingDatabaseConfigException("your message here")
```

Reply:
valuable suggestion, but this then should apply to all usage of database config. I make a new branch for these improvements to prevent blowing up the scope here okay?

Reply:
Sure, but in the meantime, using ValueError/RuntimeError instead of exception is still a good idea. (and easy)

Reply:
Done.